tracing-subscriber/src/registry/sharded.rs

```diff
@@ -47,6 +48,7 @@ use tracing_core::{
 #[derive(Debug)]
 pub struct Registry {
     spans: Slab<DataInner>,
+    current_spans: ThreadLocal<RefCell<SpanStack>>,
```

Comment:
I wonder if it's worth experimenting with a [`CachedThreadLocal`](https://amanieu.github.io/thread_local-rs/thread_local/struct.CachedThreadLocal.html) here? `CachedThreadLocal` has an added fast path when accessed by the thread that created the first value, allowing access in under 1ns for that thread. This might be a worthwhile optimization when a `Subscriber` is short-lived and only used on one thread, or in single-threaded programs.

However, I don't really know if there's additional overhead from using this, so if we did that, If we wanted to try it out, it's probably worth more benchmarking.

Reply:
I'd probably start with uncached `ThreadLocal` and see how it goes? I don't have a good enough environment to benchmark this change without producing too much noise that ultimately skew the results. if you can run it on your desktop, that might be best!

Reply:
Yup, we can look at this later, it was just a thought.